### PR TITLE
[Credentialing System] Database 

### LINKED ISSUE #1963: Reputation Oracle - Create Database Entities for Credential and Credential Validations
**Description:**
Create database entities for storing credential and credential validation information. This will involve designing and implementing database schema objects that represent credentials and their validation status in the credentialing system.

**Motivation:**
- **Structured Data Storage:** Having defined entities for credentials and their validations will enable the system to store and retrieve data in a structured and efficient way.

**Implementation Details:**
1. **Database Entity Definitions:**
   - **Credential:**
     - `id`: Integer, primary key
     - `validator_id`: Foreign key to Users table referencing the entity that created the credential
     - `reference`: String, unique, representing a credential's unique identifier
     - `description`: String, description of the credential
     - `url`: String, url where the users should go to validate their credentials
     - `status`: String, status of the validation (`ACTIVE`, `EXPIRED`)
     - `starts_at`: Datetime, date from which the credential becomes accessible for verification
     - `expires_at`: Datetime, date from which the credential becomes inaccessible for verification (nullable)
     - `created_at`: Datetime, timestamp of when the credential was created
     - `updated_at`: Datetime, timestamp of when the credential was last updated
     
   - **CredentialValidation:**
     - `id`: Integer, primary key
     - `credential_id`: Foreign key referencing `Credential.id`
     - `status`: String, status of the validation (`VALIDATED`, `ON_CHAIN`)
     - `user_id`: Foreign key to Users table referencing the user whose credential is validated.
     - `certificate`: String, signed message that validates the credential (nullable)
     - `created_at`: Datetime, timestamp of when the validation was created
     - `updated_at`: Datetime, timestamp of when the validation was last updated

2. **Database Schema Changes:**
   - Implement the schema changes to create the `Credential` and `Credential Validation` entities in the database.
   - Set appropriate indexing and constraints for the fields to ensure data consistency and retrieval performance.

### PR DESCRIPTION
**Pull request by mrhouzlane**
## Description

- Design schema for credentialing system  

## Summary of changes

-  Add entity, service, DTO for the credential system. 
- Generate migrations 

## Related issues

Closes #1963 #1960

### CODE REVIEW

File: packages/apps/reputation-oracle/server/src/modules/web3/web3.service.ts [hunk 1]
```diff
@@ -81,23 +81,37 @@ export class Web3Service {
...
       default:
-        throw new BadRequestException('Type not allowed');
+        throw new BadRequestException('Unsupported signature type');
     }
 
     return {
       from: address,
-      to: this.getOperatorAddress(),
-      contents: content,
+      to: address,
```
Comment (flopez7) [inline packages/apps/reputation-oracle/server/src/modules/web3/web3.service.ts:113]: Why did you remove `this.getOperatorAddress()`?

**Review by flopez7 — CHANGES_REQUESTED** [review]

File: packages/apps/reputation-oracle/server/src/modules/credentials/credential.entity.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { Column, Entity, ManyToOne } from 'typeorm';
+import { NS } from '../../common/constants';
+import { BaseEntity } from '../../database/base.entity';
+import { CredentialStatus } from '../../common/enums/credential';
+import { UserEntity } from '../user/user.entity';
+
+@Entity({ schema: NS, name: 'credential' })
```
Comment (flopez7) [inline packages/apps/reputation-oracle/server/src/modules/credentials/credential.entity.ts:7]: Can you set the name of the table in plural? To match the style of the other tables.

Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/credentials/credential.entity.ts]: done

File: packages/apps/reputation-oracle/server/src/modules/credentials/credential.entity.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  public status: CredentialStatus;
+
+  @Column({ type: 'timestamptz' })
+  public startsAt: Date;
+
+  @Column({ type: 'timestamptz', nullable: true })
+  public expiresAt?: Date;
+}
+
+@Entity()
```
Comment (flopez7) [inline packages/apps/reputation-oracle/server/src/modules/credentials/credential.entity.ts:31]: You should specify here the options for this entity, By the way, can you check it? I ran the script but this table is not being created

Comment (flopez7) [reply packages/apps/reputation-oracle/server/src/modules/credentials/credential.entity.ts:31]: I see, as you don't specify the schema, the table is being created in public schema instead of hmt

Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/credentials/credential.entity.ts:31]: Added Entity configuration options : `@Entity({ schema: NS, name: 'credentials_validation' })`

 Is `credentials_validation` a correct plural name for this entity  ?

Comment (flopez7) [reply packages/apps/reputation-oracle/server/src/modules/credentials/credential.entity.ts]: I would say `credential_validations` as you have multiple validations in that table

Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/credentials/credential.entity.ts]: I generated the migration based on those changes and applied it.

File: packages/apps/reputation-oracle/server/src/database/migrations/1715601874032-CredentialsSetup.ts
Comment (flopez7) [inline packages/apps/reputation-oracle/server/src/database/migrations/1715601874032-CredentialsSetup.ts:1]: Are you cleaning the previous changes in the database when creating the new ones? I miss some changes in this file

**Review by flopez7 — CHANGES_REQUESTED** [review]

**Review by flopez7 — APPROVED** [review]

**Review by portuu3 — APPROVED** [review]
